src/arrayMethodSort.js

```diff
@@ -4,8 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let callBack = compareFunction;
+
+    if (!callBack) {
+      callBack = function (a, b) {
+        if (String(a) > String(b)) {
+          return 1;
+        } else if (String(a) === String(b)) {
+          return 0;
+        } else {
+          return -1;
+        }
+      };
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (callBack(this[i], this[j]) > 0) {
+          const backup = this[i];
+
+          this[i] = this[j];
+          this[j] = backup;
+        }
+      }
+    }
```

Comment:
You are using the bubble sort algorithm, which is a simple sorting algorithm but not the most efficient one. For small arrays, it's fine. However, for larger arrays, it can be quite slow. Consider studying more efficient sorting algorithms like quicksort or mergesort for future tasks. But for this task, it's acceptable.